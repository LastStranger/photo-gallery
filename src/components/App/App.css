@import '../../common_styles/reset.css';

body {
  background-color: white;
  font-family: San Francisco, Helvetica Neue, Helvetica, sans-serif;
}

* {
  box-sizing: border-box;
}

.app-container {
  transition: filter 1s ease;

  &.is-loading {
    filter: blur(15px) grayscale(1);
    transition: filter 1s ease;
  }
}
