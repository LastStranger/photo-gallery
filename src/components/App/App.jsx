// @flow

import { h, Component } from 'preact';
import Router from 'preact-router';
import PhotoStream from '../PhotoStream';
import PhotoView from '../PhotoView';
import PopOver from '../PopOver';
import './App.css';

const POPOVER_TIMER = 1000;

class App extends Component {
  state = {
    loading: true,
    showPopOver: true
  };
  componentWillUnmount() {
    if (this.intervalID) {
      clearTimeout(this.intervalID);
    }
  }
  setLoading = (loading: boolean) => {
    if (loading) {
      this.setState({ showPopOver: true }, () => {
        setTimeout(() => this.setState({ loading: true }), 50);
      });
      if (this.intervalID) {
        clearTimeout(this.intervalID);
        this.intervalID = null;
      }
    } else {
      this.setState({ loading: false });
      this.intervalID = setTimeout(() => {
        this.setState({ showPopOver: false });
        this.intervalID = null;
      }, POPOVER_TIMER + 20);
    }
  };
  intervalID = null;
  render() {
    return (
      <section>
        {this.state.showPopOver ? <PopOver show={this.state.loading} /> : null}
        <div className={`app-container${this.state.loading ? ' is-loading' : ''}`}>
          <Router>
            <PhotoStream path="/" setLoading={this.setLoading} />
            <PhotoView path="/:id" id="n/a" setLoading={this.setLoading} />
          </Router>
        </div>
      </section>
    );
  }
}

export default App;
