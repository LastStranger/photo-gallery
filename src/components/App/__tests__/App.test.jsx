// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import App from '../';

test('renders', () => {
  const string = render(<App />);
  expect(string).toMatchSnapshot();
});
