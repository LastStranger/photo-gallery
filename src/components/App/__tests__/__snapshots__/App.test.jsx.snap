// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<section>
  <div class="popover is-visibile" role="dialog" aria-labelledby="dialogTitle">
    <div class="popover-container">
      <h2 class="popover-title" id="dialogTitle">Loading</h2>
      <img class="popover-img" src="/assets/icons/loading.svg" alt="loading animation"
      />
    </div>
  </div>
  <div class="app-container is-loading">
    <section role="switch" tabIndex="0" class="photo-view">
      <div class="header-container is-invisibile">
        <header class="header-bar header-bar-inverse">
          <h1 class="header-title"></h1>
        </header>
      </div>
      <div class="photo-viewer"></div>
      <nav class="navbar is-invisibile">
        <img class="navbar-item" src="/assets/icons/plusOne.jpg" alt="give +1"
        />
        <img class="navbar-item" src="/assets/icons/comment.jpg" alt="comment"
        />
        <img class="navbar-item" src="/assets/icons/add.jpg" alt="add" />
        <img class="navbar-item" src="/assets/icons/arrow.jpg" alt="share" />
      </nav>
    </section>
  </div>
</section>
`;
